pos_frequency_pipeline.py

```diff
@@ -2,7 +2,57 @@
 Implementation of POSFrequencyPipeline for score ten only.
 """
 
+from collections import defaultdict
+import json
+import os
+import re
+
+from pymystem3 import Mystem
+
+from constants import ASSETS_PATH
+from visualizer import visualize
+
 
 class POSFrequencyPipeline:
-    def __init__(self, assets):
-        pass
+    def __init__(self, corpus_manager):
+        self.corpus_manager = corpus_manager
+        self._current_article = None
+
+    def run(self):
+        """
+        Runs pipeline process scenario
+        """
+        articles = self.corpus_manager.get_articles()
+        for article in articles.values():
+            article.text = article.get_raw_text()
+            self._current_article = article
+            self._process()
+            self._save_meta()
+
+            png_path = os.path.join(ASSETS_PATH, f'{self._current_article.article_id}_image.png')
+            visualize(statistics=self._current_article.pos_tags,
+                      path_to_save=png_path)
+
+    def _process(self):
+        self._current_article.pos_tags = defaultdict(int)
+        text = self._current_article.text
+        text = ' '.join(re.findall(r'\w+', text.lower()))
+        result = Mystem().analyze(text)
```

Comment:
this pipeline cannot analyze, it works on top of processed dataset

Reply:
fixed